control-plane/pkg/reconciler/broker/namespaced_broker.go

```diff
@@ -112,9 +117,56 @@ func (r *NamespacedReconciler) FinalizeKind(ctx context.Context, broker *eventin
 
 	r.IPsLister.Unregister(types.NamespacedName{Namespace: broker.Namespace, Name: broker.Name})
 
+	err := r.cleanUpClusterScopedResources(ctx, broker)
+	if err != nil {
+		// we don't let this error propagate, as we don't want to requeue this forever
```

Comment:
Maybe we should propagate and requeue the finalization of the Broker so that we can try deleting the cluster scoped resources again.

If controller crashes right before this step, we will still leak cluster scoped resources...

Reply:
I went with it